frog/imports/api/activities.js

```diff
@@ -2,49 +2,146 @@
 import { Meteor } from 'meteor/meteor';
 import { Mongo } from 'meteor/mongo';
 import { omitBy, isNil } from 'lodash';
-import { uuid, type MongoT, type DashboardDataDbT } from 'frog-utils';
+import {
+  chainUpgrades,
+  uuid,
+  type MongoT,
+  type DashboardDataDbT
+} from 'frog-utils';
 
-import { operatorTypesObj } from '../operatorTypes';
-import { Graphs } from './graphs';
-import { Products } from './products';
-import { Sessions } from './sessions';
+import { activityTypesObj } from '/imports/activityTypes';
 
 export const Activities = new Mongo.Collection('activities');
-export const Operators = new Mongo.Collection('operators');
 export const Connections = new Mongo.Collection('connections');
-export const ExternalOperators = new Mongo.Collection('external_operators');
 export const DashboardData: MongoT<DashboardDataDbT> = new Mongo.Collection(
   'dashboard_data'
 );
 
+export const insertActivityMongo = (activity: Object) => {
+  try {
+    const newAct = {
+      ...activity,
+      data: activityTypesObj[activity.activityType].upgradeFunctions
+        ? chainUpgrades(
+            activityTypesObj[activity.activityType].upgradeFunctions,
+            activity.configVersion === undefined ? 1 : activity.configVersion,
+            activityTypesObj[activity.activityType].configVersion
+          )(activity.data)
+        : activity.data,
+      configVersion: activityTypesObj[activity.activityType].configVersion
+    };
+    Activities.insert(newAct);
```

Comment:
For now it is called in :
- addActivity(): called when I pull an activity from the remote library where it is necessary to update the config. But this function is also called in 2-3 other places where it might not be relevant so I could create a separate
method for remoteActivities instead of using addActivity
- duplicateActivity(): here I agree, it's probably not relevant except if we later on want to be able to clone an activity in the library without pulling the activity down, I don't know. I could probably use the normal insert
- addGraph(): here I need it as it update the configs of each activity when we import a graph (from postgrest and from computer). The only thing that has been done previously is to update the global graphVersion